File: src/main/java/control/BasicAlertController.java
package control;

import app.AppSettings;
import javafx.fxml.Initializable;
import javafx.scene.control.Button;
import javafx.scene.control.Label;
import javafx.stage.Stage;

import java.net.URL;
import java.util.ResourceBundle;

public class BasicAlertController implements Initializable {

    protected boolean accepted = false;

    public Label message;
    public Button acceptButton;

    private final String text;

    public BasicAlertController(String text){
        this(text, false);
    }

    public BasicAlertController(String text, boolean defaultState){
        this.text = text;
        accepted = defaultState;
    }

    @Override
    public void initialize(URL url, ResourceBundle resourceBundle) {
        message.setText(text);
        acceptButton.setText(AppSettings.language().getItem("acceptButton"));

        acceptButton.setOnAction(event -> {
            accepted = true;
            event.consume();
            ((Stage) acceptButton.getScene().getWindow()).close();
        });
    }

    public boolean accepted(){
        return accepted;
    }

    public boolean rejected(){
        return !accepted();
    }
}

File: src/main/scala/control/BasicAlertController.scala
package control

import java.net.URL
import java.util.ResourceBundle

import app.AppSettings
import javafx.fxml.{FXML, Initializable}
import javafx.scene.control.{Button, Label}
import javafx.stage.Stage

class BasicAlertController(text: String, defaultState: Boolean = false) extends Initializable {

    protected var _accepted: Boolean = defaultState

    @FXML var message: Label = _
    @FXML var acceptButton: Button = _

    //For Java interoperability transparency
    def this(text: String) = this(text, false)

    override def initialize(url: URL, resourceBundle: ResourceBundle): Unit = {
        message.setText(text)
        acceptButton.setText(AppSettings.language.getItemOrElse("acceptButton", "Accept"))

        acceptButton.setOnAction(event => {
            _accepted = true
            event.consume()
            acceptButton.getScene.getWindow.asInstanceOf[Stage].close()
        })
    }

    def accepted: Boolean = _accepted

    def rejected: Boolean = !accepted
}
